Weekly_Work/Week2/Metrics_Root_class.py

```diff
@@ -0,0 +1,98 @@
+import json
+import datetime
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+class Metric():
+    
+    def __init__(self, path, source_code_exclude_list=None):
+
+        self.source_code_exclude_list = source_code_exclude_list
+        self.repo_urls = set()
+
+        data_list = list()
+        self.raw_df = pd.DataFrame()
+
+        with open(path, 'r') as raw_data:
+            for line in raw_data:
+                line = json.loads(line)
+                self.repo_urls.add(line['origin'])
+
+                data_list.append(line)
+
+            self.raw_df = pd.DataFrame(data_list)
+
+
+    def _is_source_code(self, commit):
```

Comment:
I think this should be an external function, or maybe class (in case we want to maintain state). Don't you find it useful that way?

Reply:
Done! I created a class for it.

Reply:
Thx!